hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ReconTaskControllerImpl.java

```diff
@@ -251,9 +266,14 @@ private List<String> processTaskResults(List<Future<Pair<String, Boolean>>>
       if (!f.get().getRight()) {
         LOG.info("Failed task : {}", taskName);
         failedTasks.add(f.get().getLeft());
+        taskStatusCounter.updateCounter(taskName, false);
+        LOG.info("Task Name: {}, Counts: {}", taskName, taskStatusCounter.getTaskStatusCounts(taskName));
+        storeLastCompletedTransaction(taskName, events.getLastSequenceNumber(), 1);
       } else {
         taskFailureCounter.get(taskName).set(0);
-        storeLastCompletedTransaction(taskName, events.getLastSequenceNumber());
+        taskStatusCounter.updateCounter(taskName, true);
+        LOG.info("Task Name: {}, Counts: {}", taskName, taskStatusCounter.getTaskStatusCounts(taskName));
+        storeLastCompletedTransaction(taskName, events.getLastSequenceNumber(), -1);
```

Comment:
This is success condition, so again status code incorrect as -1

Reply:
Addressed in [90c1997](https://github.com/apache/ozone/pull/7517/commits/90c1997d1a1c2906198d2117542ba2fa10869a56)